### PR TITLE
Implement mint login

### PR DESCRIPTION
**Pull request by kylekthompson**
Apologies for the size of this one. It was more manageable until the point I added tests- they're _super_ verbose and Go doesn't (at least immediately obviously to me) offer many affordances to tidy them up further.

This first pass of `mint login` coordinates with RWX Cloud to acquire a device code, open that in an engineer's browser, let them authorize, and then with that authorization generate a personal access token which we store on the device. Currently, we're storing in plaintext at `~/.mint/accesstoken`, but in the future it's likely we'll introduce something like https://github.com/99designs/keyring to support storing in credential providers (keychain, pass, etc). 

https://github.com/rwx-research/mint-cli/assets/6563105/c1882e38-5108-4569-883b-c9b19488709f

### CODE REVIEW

File: cmd/mint/login.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		Use:   "login [flags]",
+	}
+)
+
+func init() {
+	// A different host can only be set over the environment
+	mintHost = os.Getenv("MINT_HOST")
+	if mintHost == "" {
+		mintHost = "cloud.rwx.com"
+	}
```
Comment (TAGraves) [inline cmd/mint/login.go:68]: I think Pierre just made a change which sets this up in root.cmd, so this might be unnecessary now

Comment (kylekthompson) [reply cmd/mint/login.go]: Yea, I had copied over some of his WIP to my branch originally but in the very complicated rebase, I missed this. Thanks!

File: cmd/mint/run.go [hunk 2]
```diff
@@ -87,17 +87,11 @@ var (
 
 		},
 		Short: "Start a new run on Mint",
-		Use:   "run [flags] --user-access-token=<token> [task]",
+		Use:   "run [flags] --access-token=<token> [task]",
```
Comment (TAGraves) [inline cmd/mint/run.go:90]: I feel like --access-token shouldn't be in this example. Everyone should be running this after running `mint login` and the helptext should take care of explaining how to set an access token manually if you're in CI

Comment (kylekthompson) [reply cmd/mint/run.go:90]: `--access-token` is also still acceptable (and in some use cases, like in CI, potentially preferable). I think we probably just don't want to leave it here as an expected/seemingly required flag

Comment (TAGraves) [reply cmd/mint/run.go:90]: Right, it should be a flag listed in the helptext, it just shouldn’t be in the canonical example. Totally fine to adjust this as a follow up. We should also add an error telling you to run mint login if no access token is set.

Comment (kylekthompson) [reply cmd/mint/run.go:90]: > We should also add an error telling you to run mint login if no access token is set.

Let's chat about this a little offline

**Review by TAGraves — APPROVED** [review]
Nice work on this!

File: internal/cli/service.go [hunk 3]
```diff
@@ -121,6 +124,69 @@ func (s Service) InitiateRun(cfg InitiateRunConfig) (*client.InitiateRunResult,
...
+	})
+	if err != nil {
+		return errors.Wrap(err, "unable to obtain an auth code")
+	}
+
+	// we print a nice message to handle the case where opening the browser fails, so we ignore this error
+	cfg.OpenUrl(authCodeResult.AuthorizationUrl) //nolint:errcheck
+
+	fmt.Fprintln(cfg.Stdout)
+	fmt.Fprintln(cfg.Stdout, "To authorize this device, you'll need to login to RWX Cloud and choose an organization.")
```
[the remark this answers is not in the document]
Comment (kylekthompson) [reply internal/cli/service.go:147]: Correct. And we probably have some experience things to sort out in the >1 case anyway (PATs are only good for one org right now, so you have to login again for another and that will remove your other creds)

**Pull request merged**